src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -406,6 +666,52 @@ CHIP_ERROR DefaultSceneTableImpl::RemoveSceneTableEntryAtPosition(FabricIndex fa
     return fabric.RemoveScene(mStorage, scene.mStorageId);
 }
 
+CHIP_ERROR DefaultSceneTableImpl::GetAllSceneIdsInGroup(FabricIndex fabric_index, GroupId group_id, Span<SceneId> & scene_list)
+{
+    FabricSceneData fabric(fabric_index, mMaxScenesPerFabric, mMaxScenesGlobal);
+    SceneTableData scene(fabric_index);
+
+    auto * iterator     = this->IterateSceneEntries(fabric_index);
+    SceneId * list      = scene_list.data();
+    uint8_t scene_count = 0;
+
+    while (iterator->Next(scene) && scene_count < scene_list.size())
+    {
+        if (scene.mStorageId.mGroupId == group_id)
+        {
+            list[scene_count] = scene.mStorageId.mSceneId;
+            scene_count++;
+        }
+    }
+    scene_list.reduce_size(scene_count);
+
+    // Check that the buffer was big enough for all scenes within this group
+    bool more_scenes_left = iterator->Next(scene);
+    iterator->Release();
+    VerifyOrReturnError(more_scenes_left == false, CHIP_ERROR_BUFFER_TOO_SMALL);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultSceneTableImpl::DeleteAllScenesInGroup(FabricIndex fabric_index, GroupId group_id)
+{
+    VerifyOrReturnError(IsInitialized(), CHIP_ERROR_INTERNAL);
+
+    FabricSceneData fabric(fabric_index, mMaxScenesPerFabric, mMaxScenesGlobal);
+    SceneTableData scene(fabric_index);
+
+    ReturnErrorOnFailure(fabric.Load(mStorage));
+
+    for (uint8_t i = 0; i < mMaxScenesPerFabric; i++)
+    {
+        if (fabric.scene_map[i].mGroupId == group_id)
+        {
+            ReturnErrorOnFailure(fabric.RemoveScene(mStorage, fabric.scene_map[i]));
```

Comment:
I assume this RemoveScene call does not change `fabric.scene_map`?  Worth documenting that, if so.